server/BUILD.bazel

```diff
@@ -58,30 +55,32 @@ esbuild(
         "vscode-languageserver-textdocument",
         "vscode-html-languageservice",
     ],
-    config = "esbuild.mjs",
+    format = "cjs",
     # Do not enable minification. It seems to break the extension on Windows (with WSL). See #1198.
     minify = False,
+    platform = "node",
+    sourcemap = "external",
     visibility = [
         "//integration:__subpackages__",
-    ]
+    ],
 )
 
 expand_template(
     name = "package_json_expanded",
-    template = "package.json",
     out = "package_expanded.json",
-    substitutions = {
-        "0.0.0-PLACEHOLDER": VERSION,
-    }
+    stamp_substitutions = {
+        "0.0.0-PLACEHOLDER": "{{BUILD_SCM_VERSION}}",
+    },
+    template = "package.json",
```

Comment:
FYI: Mostly the `stamp_substitutions` are relevant for review. I kept the buildifier reformatting as they shouldn't hurt.